src/openms/source/CHEMISTRY/ResidueDB.cpp

```diff
@@ -468,36 +388,41 @@ namespace OpenMS
     return rs;
   }
 
-  void ResidueDB::buildResidueNames_()
+  void ResidueDB::buildResidueName_(Residue* r)
```

Comment:
argument should be `const Residue* const r)`

Reply:
I think it currently needs Residue* to store it in
    std::set<Residue*> residues_;
and
    std::set<const Residue*> const_residues_;

probably because it provides a:
inline ResidueIterator beginResidue() { return residues_.begin(); }
inline ResidueIterator endResidue() { return residues_.end(); }

might be a good idea to see where these are used and get rid of them (including the non-const stored residues)